plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/features/core/model/SourceModelStorage.java

```diff
@@ -15,27 +15,39 @@
  ********************************************************************************/
 package org.eclipse.glsp.server.features.core.model;
 
+import org.eclipse.glsp.server.actions.SaveModelAction;
 import org.eclipse.glsp.server.utils.ClientOptionsUtil;
 
 /**
- * A source model loader loads models into the model state.
+ * A source model storage handles the persistence of source models, i.e. loading and saving it from/into the model
+ * state.
  * <p>
  * A <i>source model</i> is an arbitrary model from which the graph model of the diagram is to be created.
- * Implementations of source model loaders are specific to the type of source model or persistence format that is used
- * for a type of source model. A source model loader obtains the information on which source model shall loaded from a
+ * Implementations of source model stores are specific to the type of source model or persistence format that is used
+ * for a type of source model. A source model storage obtains the information on which source model shall loaded from a
  * {@link RequestModelAction}; typically its client options. Once the source model is loaded, a model loader is expected
  * to put the loaded source model into the model state for further processing, such as transforming the loaded model
  * into a graph model (see {@link GModelFactory}).
+ * On {@link #saveSourceModel(SaveModelAction) save}, the source model storage persists the source model from the model
+ * state.
  * </p>
  *
  * @see ClientOptionsUtil
  * @see GModelFactory
  */
-public interface ModelSourceLoader {
+public interface SourceModelStorage {
    /**
     * Loads a source model into the <code>modelState</code>.
     *
     * @param action Action sent by the client to specifying the information needed to load the source model.
     */
    void loadSourceModel(RequestModelAction action);
+
+   /**
+    * Saves the source model from the <code>modelState</code>.
+    *
+    * @param action Action sent by the client to specifying the information needed to save the source model.
+    * @return Returns <code>true</code> if save is done, <code>false</code> otherwise.
```

Comment:
Technically the `GModelJsonStorage` implementation does not conform to this convention.
It either returns `true` or throws a `GLSPServerException`. `false` is never returned.
I'm wondering wether a boolean return value makes sense in this case.
I'd opt for `void` return and document that the method is expected to thrown an excpetion if something went wrong during the save process.

Reply:
Agreed, I updated the code and javadoc accordingly.